utils/build.ps1

```diff
@@ -740,10 +736,43 @@ function Fetch-Dependencies {
     }
   }
 
+  function Ensure-PythonModules($Python) {
+    # First ensure pip is installed, else bootstrap it
+    try {
+      Invoke-Program -OutNull $Python -m pip *> $null
+    } catch {
+      Write-Output "Installing pip ..."
+      Invoke-Program -OutNull $Python '-I' -m ensurepip -U --default-pip
+    }
```

Comment:
Is there a way to just perform the installation assuming it is not present without any adverse effects?

Reply:
I don’t think that’s a good idea, but it’s certainly doable.

Reply:
Had a simple try and found that we cannot easily reuse `Extract-ZipFile` because `Expand-Archive` requires `.zip` as file extension, and wheels aren’t extracted to standalone directories. Given that `pip` is already included (no download required) and simply just not installed, it shouldn’t be a real problem.

Reply:
I think that using `pip` is fine, just trying to see if we can simplify some of the logic here by just unconditionally doing `python -I -m ensurepip -U --default-pip`.

Reply:
Running conditionally should work, but I intended to log precisely about what’s happening. Also need to notice that this command will output some noise when you already have pip installed.

Reply:
Thanks for working on this @stevapple! The move away from host Python solved an issue for me accessing modules in LLDB tests. And I can re-use your approach for installing modules: https://github.com/swiftlang/swift/pull/76894/commits/942538fbfcee4f68199503e88d6e31364c668cd8 What else do you think will be necessary to merge this PR?

Reply:
@weliveindetail Glad to hear that it helped you! Let’s wait for @compnerd to kick off a new CI run and see if there’s any other problem.

Reply:
Great. Then I will wait for it to land, rebase my changes on top of it and add my Python modules for lldb tests. Maybe I will have to factor out a function or so, that would be no problem to do. What do you think @compnerd?